test/unit/view-layout/test-multiview.cpp

```diff
@@ -40,9 +40,55 @@ TYPED_TEST(MultiViewUnitTest, Constructors)
   a1[0] = val;
   a2[0] = val;
 
+  TypeParam   b1[10] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+  TypeParam   b2[10] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 0};
+  TypeParam const * data2[2];
+
+  data2[0] = b1;
+  data2[1] = b2;
+
+  b1[0] = val;
+  b2[0] = val;
+
   RAJA::MultiView<TypeParam, layout> view(data, layout(10));
   ASSERT_EQ( val, view(0,0) );
 
+  /*
+   * Should be able to construct a const MultiView from const array of arrays
+   */
+  RAJA::MultiView<TypeParam const, layout> view_from_const(data2, layout(10));
+  ASSERT_EQ( val, view_from_const(0,0) );
+
+  /*
+   * Should be able to construct an empty const MultiView
+   */
+  RAJA::MultiView<TypeParam const, layout> view_from_const2;
+  ASSERT_EQ( nullptr, view_from_const2.get_data() );
+  ASSERT_EQ( 1, (view_from_const2.get_layout()).size() );
+
+  /*
+   * Should be able to set data and layout in an empty const MultiView
+   */
+  view_from_const2.set_layout(layout(10));
+  view_from_const2.set_data(data2);
+  ASSERT_EQ( 10, (view_from_const2.get_layout()).size() );
+  ASSERT_EQ( val, view_from_const2(0,0) );
```

Comment:
You could add another test like this but with data instead of data2 to ensure views to const work with pointers to non-const.